commands/setbit.md

```diff
@@ -30,3 +30,118 @@ SETBIT mykey 7 1
 SETBIT mykey 7 0
 GET mykey
 ```
+
+## Pattern: accessing the entire bitmap
+
+There are cases when you need to set all the bits of single bitmap at once, for
+example when initializing it to a default non-zero value. It is possible to do
+this with multiple calls to the `SETBIT` command, one for each bit that needs to
+be set. However, as an optimization, you can use a single `SET` command to set
+the entire bitmap.
+
+Bitmaps are not an actual data type, but a set of bit-oriented operations
+defined on the String type (for more information refer to the
+[Bitmaps section of the Data Types Introduction page][ti]). This means that
+bitmaps can be used with in string command, and most importantly with `SET` and
+`GET`.
+
+Because Redis' strings are binary-safe, a bitmap is trivially encoded as a bytes
+stream. The first byte of the string corresponds to offsets 0..7 of
+the bitmap, the second byte to the 8..15 range, and so forth.
+
+For example, after setting a few bits, getting the string value of the bitmap
+would look like this:
+
+```
+> SETBIT bitmapsarestrings 2 1
+> SETBIT bitmapsarestrings 3 1
+> SETBIT bitmapsarestrings 5 1
+> SETBIT bitmapsarestrings 10 1
+> SETBIT bitmapsarestrings 11 1
+> SETBIT bitmapsarestrings 14 1
+> GET bitmapsarestrings
+"42"
+```
+
+By getting the string representation of a bitmap, the client can then parse the
+response's bytes by extracting the bit values using native bit operations in its
+native programming language. Symmetrically, it is also possible to set an entire
+bitmap by performing the bits-to-bytes encoding in the client and calling `SET`
+with the resultant string.
+
+[ti]: /topics/data-types-intro#bitmaps
+
+## Advanced Pattern: accessing bitmap ranges
```

Comment:
This is an interesting pattern (wish I knew about the `bit` object in Lua a few months ago) however I'm wondering if this is more of Lua pattern than something that belongs on a the `SETBIT` page.

Reply:
This is indeed Lua, with the added advantage that it can be run on the server. However, at the same time, it serves as an implementation example that can be ported elsewhere. The same could have been in Python (but no Python in the docs), Ruby (meh, and I don't really know it), JS (...) or pseudo code.

Going forward, I think I'll continue the use of Lua for examples for the above reasons.